server/pulp/server/db/querysets.py

```diff
@@ -135,3 +137,43 @@ def get_repo_or_missing_resource(self, repo_id):
             return self.get(repo_id=repo_id)
         except DoesNotExist:
             raise pulp_exceptions.MissingResource(repository=repo_id)
+
+
+class RepositoryContentUnitQuerySet(CriteriaQuerySet):
+    """
+    Custom queryset for repository content units.
+    """
+
+    def _num_between(self, field, start=None, end=None, repo_id=None):
+        """
+        Helper function to query units of based on timestamps.
+        """
+
+        if end is None:
```

Comment:
Consider replacing L#152 - L#165 with a simpler version using operator.reduce and a list of `Q()` objects.

Here is [an example](https://github.com/pulp/pulp_rpm/blob/7715f91c7488a0d34b74174fd5e4e485b0251e2b/plugins/pulp_rpm/plugins/importers/yum/purge.py#L281-L282) that I made.